.ci/cico_minikube_shost.sh

```diff
@@ -0,0 +1,197 @@
+    oc create namespace che
+  fi
+}
+
+# Deploy Eclipse Che in single host mode(gateway exposure type)
+function runSHostGatewayExposure() {
+    # Patch file to pass to chectl
+    cat >/tmp/che-cr-patch.yaml <<EOL
+spec:
+  server:
+    serverExposureStrategy: 'single-host'
+  auth:
+    updateAdminPassword: false
+    openShiftoAuth: false
+  k8s:
+    singleHostExposureType: 'gateway'
+EOL
+    echo "======= Che cr patch ======="
+    cat /tmp/che-cr-patch.yaml
+
+    # Use custom changes, don't pull image from quay.io
+    checkNamespace
+    cat ${OPERATOR_REPO}/deploy/operator.yaml | \
+    sed 's|imagePullPolicy: Always|imagePullPolicy: IfNotPresent|' | \
+    sed 's|quay.io/eclipse/che-operator:nightly|'${OPERATOR_IMAGE}'|' | \
+    oc apply -n ${NAMESPACE} -f -
+
+    # Start to deploy Che
+    chectl server:start --platform=minikube --skip-kubernetes-health-check --installer=operator \
+        --chenamespace=${NAMESPACE} --che-operator-image=${OPERATOR_IMAGE} --che-operator-cr-patch-yaml=/tmp/che-cr-patch.yaml
+
+    # Create and start a workspace
+    getSingleHostToken # Function from ./util/ci_common.sh
+    chectl workspace:create --start --devfile=$OPERATOR_REPO/.ci/util/devfile-test.yaml
+
+    # Wait for workspace to be up
+    waitSingleHostWorkspaceStart
+
+    chectl server:delete --chenamespace=${NAMESPACE} --skip-deletion-check
+}
+
+# Deploy Eclipse Che in single host mode(native exposure type)
+function runSHostNativeExposure() {
+    # Patch file to pass to chectl
+    cat >/tmp/che-cr-patch.yaml <<EOL
+spec:
+  server:
+    serverExposureStrategy: 'single-host'
+  auth:
+    updateAdminPassword: false
+    openShiftoAuth: false
+EOL
+    echo "======= Che cr patch ======="
+    cat /tmp/che-cr-patch.yaml
+
+    # Use custom changes, don't pull image from quay.io
+    checkNamespace
+    cat ${OPERATOR_REPO}/deploy/operator.yaml | \
+    sed 's|imagePullPolicy: Always|imagePullPolicy: IfNotPresent|' | \
+    sed 's|quay.io/eclipse/che-operator:nightly|'${OPERATOR_IMAGE}'|' | \
+    oc apply -n ${NAMESPACE} -f -
+
+    # Start to deploy Che
+    chectl server:start --platform=minikube --skip-kubernetes-health-check --installer=operator \
+        --chenamespace=${NAMESPACE} --che-operator-image=${OPERATOR_IMAGE} --che-operator-cr-patch-yaml=/tmp/che-cr-patch.yaml
+
+    # Get the token from .ci/util/common.sh
+    getCheAcessToken
+    chectl workspace:create --start --devfile=$OPERATOR_REPO/.ci/util/devfile-test.yaml
+
+    # Wait for workspace to be up for native deployment from .ci/util/common.sh
+    waitWorkspaceStart
+
+    chectl server:delete --chenamespace=${NAMESPACE} --skip-deletion-check
+}
+
+init
+source "${OPERATOR_REPO}"/.ci/util/ci_common.sh
+echo "[INFO] Start to Building Che Operator Image"
+buildCheOperatorImage
+
+echo "[INFO] Start to run single host with gateway exposure mode"
+runSHostGatewayExposure
+
+echo "[INFO] Start to run single host in native mode"
+runSHostNativeExposure
```

Comment:
Two different checks is better then a single one

Reply:
Understood. I'll change!